quickwit/quickwit-config/src/storage_config.rs

```diff
@@ -374,6 +391,26 @@ pub struct FileStorageConfig;
 #[serde(deny_unknown_fields)]
 pub struct RamStorageConfig;
 
+#[derive(Debug, Clone, Default, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct GoogleCloudStorageConfig {
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub credential_path: Option<String>,
+}
+
+impl GoogleCloudStorageConfig {
+    pub const GOOGLE_CLOUD_STORAGE_CREDENTIAL_PATH_ENV_VAR: &'static str = "QW_GOOGLE_CLOUD_STORAGE_CREDENTIAL_PATH";
+
+    /// Attempts to find the credential path in the environment variable
+    /// `QW_GOOGLE_CLOUD_STORAGE_CREDENTIAL_PATH` or the config.
+    pub fn resolve_credential_path(&self) -> Option<String> {
+        env::var(Self::GOOGLE_CLOUD_STORAGE_CREDENTIAL_PATH_ENV_VAR)
```

Comment:
I think we tend to give priority to configuration, and when absent, inspect environment variables.

Reply:
Oops, I copied those logic from azure storage.

Reply:
The current logic aligns with S3 and Azure. Should I open a new issue to investigate this behavior?

Reply:
Oh ok! Let it that way. I'll open the ticket. Thanks.